src/back/glsl/mod.rs

```diff
@@ -353,6 +360,35 @@ impl<'a, W: Write> Writer<'a, W> {
         let mut namer = proc::Namer::default();
         namer.reset(module, keywords::RESERVED_KEYWORDS, &["gl_"], &mut names);
 
+        // Collect the list of unused, unsized structs for this entrypoint
+        let mut unused_unsized_types = Vec::default();
+        for (handle, global) in module.global_variables.iter() {
+            if info.get_entry_point(ep_idx)[handle].is_empty() {
+                match module.types[global.ty].inner {
+                    // If the global type is an unsized array
+                    crate::TypeInner::Array { size, .. } => {
+                        if crate::ArraySize::Dynamic == size {
+                            unused_unsized_types.push(global.ty);
+                        }
+                    }
+                    // Or if it is a struct with an unsized array as the last element
+                    crate::TypeInner::Struct { ref members, .. } => {
+                        if let Some(last) = members.last() {
+                            if let crate::TypeInner::Array { size, .. } =
+                                module.types[last.ty].inner
+                            {
+                                if crate::ArraySize::Dynamic == size {
+                                    unused_unsized_types.push(global.ty);
+                                    unused_unsized_types.push(last.ty);
```

Comment:
we shouldn't add the contained type to the list, we only care about the global types here.